src/commonMain/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumClient.kt

```diff
@@ -199,15 +187,17 @@ class ElectrumClient(
 
     private val json = Json { ignoreUnknownKeys = true }
 
+    // channel to send and receive request/responses with the server
     private val eventChannel: Channel<ClientEvent> = Channel(BUFFERED)
     private var output: Channel<ByteArray> = Channel(BUFFERED)
 
+    // connection status to the server
     private val _connectionState = MutableStateFlow<Connection>(Connection.CLOSED(null))
     val connectionState: StateFlow<Connection> get() = _connectionState
 
-    private val notificationsChannel = BroadcastChannel<ElectrumMessage>(BUFFERED)
-
-    fun openNotificationsSubscription() = notificationsChannel.openSubscription()
+    // subscriptions notifications (headers, script_hashes, etc.)
+    private val _notifications = MutableSharedFlow<ElectrumMessage>(replay = 0, extraBufferCapacity = BUFFERED)
```

Comment:
@sstone 

This follows Kotlin [migration's doc](https://github.com/Kotlin/kotlinx.coroutines/blob/d88a8c227a295f91251960c8fe6b21a02ddf704b/kotlinx-coroutines-core/common/src/flow/SharedFlow.kt#L100-L103):
> To migrate BroadcastChannel usage to SharedFlow, start by replacing usages of the BroadcastChannel(capacity) constructor with MutableSharedFlow(0, extraBufferCapacity=capacity) (broadcast channel does not replay values to new subscribers). Replace send and trySend calls with emit and tryEmit, and convert subscribers' code to flow operators.

The special negative capacity translated to `Int.MaxValue` in the [SharedFlow constructor](https://github.com/Kotlin/kotlinx.coroutines/blob/d88a8c227a295f91251960c8fe6b21a02ddf704b/kotlinx-coroutines-core/common/src/flow/SharedFlow.kt#L275) is a bit 😱 .

Reply:
So, the comment above wasn't accurate. There is a [check on non-negative value for `extraBufferCapacity`](https://github.com/Kotlin/kotlinx.coroutines/blob/d88a8c227a295f91251960c8fe6b21a02ddf704b/kotlinx-coroutines-core/common/src/flow/SharedFlow.kt#L270) in that constructor.

I settled with a non-zero value `extraBufferCapacity`, because doc says that the emitter will be blocked if there is no buffer and listeners are not ready. The buffer size `64` is the same as the default for [buffered channels](https://github.com/Kotlin/kotlinx.coroutines/blob/d88a8c227a295f91251960c8fe6b21a02ddf704b/kotlinx-coroutines-core/common/src/channels/Channel.kt#L732)